main.go

```diff
@@ -119,6 +119,13 @@ func main() {
 		log.Fatal("Invalid value for --prometheus_gateways")
 	}
 
+	if *systemProxy != "" {
+		_, err = utils.ParseProxyUrl(*systemProxy)
+		if err != nil {
+			log.Fatalf("Invalid value for --proxy: %v", err)
```

Comment:
This tool tries to follow a philosophy where it has to work to some extent regardless of how badly it's misconfigured. Considering that proxy list can be templated now it's better to avoid this type of check unless absolutely necessary (I plan to remove similar check for metrics as well). The check should be performed by the module that uses this value instead

Reply:
Yes, I see, but I think it's very bad when you have set proxy url and its not valid and error will be ignored even if you get some message on screen but you can not see it in time. In this case you will get a buch of outgoing traffic from your pods/container without using proxy. It's can compomate you account and you will get suspended account in cloud or light up your real ip adresses. So my opinoin if you set proxyurl but if it's have some vrong format, app should not send traffic. It may not related to the jobs config but for system variable or argument value - proxy with error handling should be I think.

Reply:
@lollypot we can probably work on separate commandline switch for that but I feel like this level of concern is close to a point where it's more viable to utilize your own config or to secure the vm/container you are running it in to forward all of the traffic through the proxy/vpn/tor instead (i.e. what's going to stop me from embedding a request into the template evaluation inside some other job? there's already a function for that used which I added to be able to retrieve proxylist from a custom location and it's not going through any proxy. it's highly unlikely for me to do something as stupid as using it for an attack but it's not impossible)

Reply:
Ok got it, thank you.